src/commands/cmd_query.c

```diff
@@ -73,21 +61,21 @@ void _MGraph_Query(void *args) {
     pthread_setspecific(_tlsASTKey, ast);
 
     // Try to access the GraphContext
-    RedisModule_ThreadSafeContextLock(ctx);
+    CommandCtx_ThreadSafeContextLock(qctx);
     GraphContext *gc = GraphContext_Retrieve(ctx, qctx->graphName);
-    RedisModule_ThreadSafeContextUnlock(ctx);
-    
+    CommandCtx_ThreadSafeContextUnlock(qctx);
```

Comment:
theadeSafeCtx should be released only after the 'if(!gc)' block to prevent edge case of two writers that create a new graph simultaneously and one of them failed

Reply:
👍